TODO.validation-gen

```diff
@@ -18,6 +18,7 @@ P0 ("things we need to do for 1.34"):
   - mutual exclusivitiy - field X is not allowed when Y
   - might require cross-field/conditional defaulting
     - might break the zero-value checking
+    - consider merging defaulter-gen with validation-gen?
```

Comment:
Just a FYI here- @yongruilin was talking with Joel Speed and https://github.com/kubernetes-sigs/kube-api-linter may need to somehow reuse/replicate the discovery pass so that is has full tag access.